jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ResourceHandlerTest.java

```diff
@@ -655,7 +660,19 @@ public void before() throws Exception
         _local.getConnectionFactory(HttpConfiguration.ConnectionFactory.class).getHttpConfiguration().setSendServerVersion(false);
         _server.addConnector(_local);
 
-        _rootResourceHandler = new ResourceHandler();
+        _rootResourceHandler = new ResourceHandler()
+        {
+            @Override
+            protected HttpContent.Factory setupHttpContentFactory()
+            {
+                // For testing the cache should be configured to validate the entry on every request.
+                HttpContent.Factory contentFactory = new ResourceHttpContentFactory(ResourceFactory.of(getBaseResource()), getMimeTypes());
+                contentFactory = new PreCompressedHttpContentFactory(contentFactory, getPrecompressedFormats());
+                contentFactory = new FileMappingHttpContentFactory(contentFactory);
+                contentFactory = new ValidatingCachingHttpContentFactory(contentFactory, 0, _local.getByteBufferPool());
+                return contentFactory;
+            }
+        };
```

Comment:
Use `ResourceHandler.setHttpContentFactory(...)` instead of subclassing.

Reply:
Is there a reason to prefer one technique over the other?   

They are slightly different in that subclassing will create the HCF during start, when all configuration is set/known, and that it will then remove that instance during stop, so a new instance is created on next start cycle, using latest configuration.    The setter will set a HCF that will be durable over stop/start cycles, so it will not see config changes in the RH.

So I kind of prefer the subclass style when the configuration of the HCF is entirely dependent on the RH.  The setter style is good for when the HCF has other external configuration.

We should have tests that use both style to make sure they work.   In these specific tests, it doesn't really matter one way or the other.

Reply:
After a hangout with Simone we decided to not add the setter method for HttpContentFactory at all as it makes the configuration more difficult.